lib/utils/docker.coffee

```diff
@@ -0,0 +1,201 @@
+# Functions to help actions which rely on using docker
+
+# Use this function to seed an action's list of capitano options
+# with the docker options. Using this interface means that
+# all functions using docker will expose the same interface
+#
+# NOTE: Care MUST be taken when using the function, so as to
+# not redefine/override options already provided.
+exports.appendOptions = (opts) ->
+	opts.concat [
+		{
+			signature: 'docker'
+			parameter: 'docker'
+			description: 'Path to a local docker socket'
+			alias: 'P'
+		},
+		{
+			signature: 'dockerHost'
+			parameter: 'dockerHost'
+			description: 'The address of the host containing the docker daemon'
+			alias: 'h'
+		},
+		{
+			signature: 'dockerPort'
+			parameter: 'dockerPort'
+			description: 'The port on which the host docker daemon is listening'
+			alias: 'p'
+		},
+		{
+			signature: 'ca'
+			parameter: 'ca'
+			description: 'Docker host TLS certificate authority file'
+		},
+		{
+			signature: 'cert'
+			parameter: 'cert'
+			description: 'Docker host TLS certificate file'
+		},
+		{
+			signature: 'key'
+			parameter: 'key'
+			description: 'Docker host TLS key file'
+		}
+	]
+
+exports.generateConnectOpts = generateConnectOpts = (opts) ->
+	connectOpts = {}
+	# Firsly need to decide between a local docker socket
+	# and a host available over a host:port combo
+	if opts.docker? and not opts.dockerHost?
+		# good, local docker socket
+		connectOpts['socketPath'] = opts.docker
+	else if opts.dockerHost? and not opts.docker?
+		# Good a host is provided, and local socket isn't
+		connectOpts['host'] = opts.dockerHost
+		connectOpts['port'] = opts.dockerPort || 2376
+	else if opts.docker? and opts.dockerHost?
+		# Both provided, no obvious way to continue
+		throw new Error("Both a local docker socket and docker host have been provided. Don't know how to continue.")
+	else
+		# None provided, assume default docker local socket
+		connectOpts['socketPath'] = '/var/run/docker.sock'
+
+	# Now need to check if the user wants to connect over TLS
+	# to the host
+
+	# If any are set...
+	if (opts.ca? or opts.cert? or opts.key?)
+		# but not all
+		if not (opts.ca? and opts.cert? and opts.key?)
+			throw new Error('You must provide a CA, certificate and key in order to use TLS')
+		connectOpts['ca'] = opts.ca
+		connectOpts['cert'] = opts.cert
+		connectOpts['key'] = opts.key
+
+	return connectOpts
+
+exports.tarDirectory = tarDirectory = (dir) ->
+	Promise = require('bluebird')
+	tar = require('tar-stream')
+	klaw = require('klaw')
+	path = require('path')
+	fs = require('mz/fs')
+	streamToPromise = require('stream-to-promise')
+
+	getFiles = ->
+		Promise.resolve(streamToPromise(klaw(dir)))
```

Comment:
I think you could avoid needing to wrap this by using [any-promise registrating: `require('any-promise/register/bluebird')`](https://github.com/kevinbeaty/any-promise#usage-with-registration)